jms/src/main/scala/akka/stream/alpakka/jms/JmsBrowseStage.scala

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (C) 2016-2018 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.stream.alpakka.jms
+
+import javax.jms._
+
+import akka.stream.{ActorAttributes, Attributes, Outlet, SourceShape}
+import akka.stream.stage.{GraphStage, GraphStageLogic, OutHandler}
+import java.util.{Enumeration => JEnumeration}
+
+private[stream] final class JmsBrowseStage(settings: JmsBrowseSettings) extends GraphStage[SourceShape[Message]] {
+
+  private val out = Outlet[Message]("JmsBrowseStage.out")
+  val shape = SourceShape(out)
+
+  override protected def initialAttributes: Attributes =
+    ActorAttributes.dispatcher("akka.stream.default-blocking-io-dispatcher")
+
+  override def createLogic(inheritedAttributes: Attributes): GraphStageLogic =
+    new GraphStageLogic(shape) with OutHandler {
+      setHandler(out, this)
+
+      var connection: Connection = _
+      var session: Session = _
+      var browser: QueueBrowser = _
+      var messages: JEnumeration[Message] = _
+
+      override def preStart(): Unit = {
+        val queue = settings.destination.getOrElse { throw new IllegalArgumentException("Destination is missing") }
```

Comment:
Check this on construction instead of materialization to fail as fast as possible